File: Data Structures & Algorithms/search-2d-matrix/submission-3.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l, r = 0 , len(matrix)* len(matrix[0])

        while l < r:
            mid = l + (r - l) // 2
            midVal = matrix[mid//len(matrix[0])][mid % len(matrix[0])]

            if midVal > target:
                r = mid
            elif midVal < target:
                l = mid+1
            else:
                return True
            
        return False

File: Data Structures & Algorithms/binary-search/submission-3.go
func search(nums []int, target int) int {
    l, r := 0, len(nums)

    mid := (l+r)/2
    for l < r {
        mid = (l+r)/2
        switch {
            case nums[mid] > target:
                r = mid
            case nums[mid] < target:
                l = mid+1
            default:
                return mid
        }
    }
        return -1

}
